### PR TITLE
Add a separate example section for escaping of characters

### PR DESCRIPTION
**Pull request by dryabov**
See merged PR https://github.com/php/php-src/pull/7420 (https://github.com/php/php-src/commit/d3a6054d4441a6d431e6ca7aa63de62cee5585c2)

### CODE REVIEW

File: reference/filesystem/functions/parse-ini-file.xml [hunk 1]
```diff
@@ -250,17 +250,66 @@ negative_two = ~1
...
 path = ${PATH}
 also_five = ${five}
 
 ]]>
     </programlisting>
    </example>
   </para>
+  <para>
+   <example>
+    <title>escaping characters</title>
```
Comment (cmb69) [inline reference/filesystem/functions/parse-ini-file.xml:262]:
```suggestion
    <title>Escaping Characters</title>
```

File: reference/filesystem/functions/parse-ini-file.xml [hunk 2]
```diff
@@ -250,17 +250,66 @@ negative_two = ~1
...
+    <para>
+     Some characters have special meaning in double-quoted strings and must be escaped by the backslash prefix.
+	 First of all, these are the double quote <code>"</code> as the boundary marker, and the backslash <code>\</code> itself
```
Comment (cmb69) [inline reference/filesystem/functions/parse-ini-file.xml:265]: Nit: indentation (also several lines further down).

File: reference/filesystem/functions/parse-ini-file.xml [hunk 3]
```diff
@@ -250,17 +250,66 @@ negative_two = ~1
...
+	 (if followed by one of the special characters):
+    </para>
+    <programlisting>
+<![CDATA[
+quoted = "She said \"Exactly my point\"." ; Results in a string with quote marks in it.
+hint = "Use \\\" to escape double quote" ; Results in: Use \" to escape double quote
+]]>
+    </programlisting>
+    <para>
+     There is an exception made for Windows-like paths: it's possible to don't escape trailing backslash
```
Comment (cmb69) [inline reference/filesystem/functions/parse-ini-file.xml:275]:
```suggestion
     There is an exception made for Windows-like paths: it's possible to not escape trailing backslash
```

File: reference/filesystem/functions/parse-ini-file.xml [hunk 4]
```diff
@@ -250,17 +250,66 @@ negative_two = ~1
...
+	 (there is one double-quoted string directly followed by another one):
+    </para>
+    <programlisting>
+<![CDATA[
+long_text = "Lorem \"ipsum\"""
+ dolor" ; Results in: Lorem "ipsum"\n dolor
+]]>
+    </programlisting>
+    <para>
+     Another character with special meaning is <code>$</code> (the dollar).
```
Comment (cmb69) [inline reference/filesystem/functions/parse-ini-file.xml:295]:
```suggestion
     Another character with special meaning is <code>$</code> (the dollar sign).
```

**Review by cmb69 — COMMENTED** [review]
Thank you very much! Besides the few nits further down, I assume this documentation is only for PHP 8.1+. If so, that should be clarified, since the manual is supposed to cover both PHP 7 and 8.

File: reference/filesystem/functions/parse-ini-file.xml (hunk 1 above)
Comment (dryabov) [reply reference/filesystem/functions/parse-ini-file.xml:262]: Previous example section ("value interpolation") has lowercase title, so let either keep both lowercase, or change pervious title to the "Proper Case" too.

File: reference/filesystem/functions/parse-ini-file.xml (hunk 2 above)
Comment (dryabov) [reply reference/filesystem/functions/parse-ini-file.xml]: Other suggestions have been merged.

**Comment by dryabov** [comment]
>  I assume this documentation is only for PHP 8.1+

No, it's a general one that works in all PHP versions (since 5.3, if I'm not mistaken). The difference between 8.0 and 8.1 is an edge case of `\"` character sequence that previously was possible to encode both as `\\"` and `\\\"`, but starting from 8.1 the only possible way is to use `\\\"` (similar to PHP's escaping rules).

File: reference/filesystem/functions/parse-ini-file.xml (hunk 1 above)
Comment (cmb69) [reply reference/filesystem/functions/parse-ini-file.xml:262]: I suggest to change "value interpolation" to "Value Interpolation", but would leave the other example titles as they are.

Comment (dryabov) [reply reference/filesystem/functions/parse-ini-file.xml]: Done

**Comment by cmb69** [comment]
Ah, fine then. Thank you!

**Pull request merged**